lib/hooks/useResources.ts

```
@@ -0,0 +1,200 @@
+  );
+
+  // Fetch resources function
+  const fetchResources = useCallback(async () => {
+    try {
+      setResourcesLoading(true);
+
+      // Fetch all modules for the selector
+      const modulesData = await api.getModules();
+      const modulesList = modulesData.modules || [];
+      setResourceModules(
+        modulesList.map((m: any) => ({
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)